### ISSUE TITLE
todo

### ISSUE THREAD

**Issue by lucidrains** [lucidrains/PEER-pytorch#1]
- [ ] batchnorm and look into the causality issue from pkm paper
- [ ] think about the GLU extension w/ triton if public reports good results
